code/game/area/areas.dm

```diff
@@ -490,3 +497,130 @@ GLOBAL_LIST_EMPTY(teleportlocs)
+						continue
+					target_turf = locate(iterating_turf.x + 1, iterating_turf.y - 1, iterating_turf.z)
+				if(SOUTHWEST_JUNCTION)
+					if(bitfield & SOUTH_JUNCTION || bitfield & WEST_JUNCTION)
+						continue
+					target_turf = locate(iterating_turf.x - 1, iterating_turf.y - 1, iterating_turf.z)
+				if(NORTHWEST_JUNCTION)
+					if(bitfield & NORTH_JUNCTION || bitfield & WEST_JUNCTION)
+						continue
+					target_turf = locate(iterating_turf.x - 1, iterating_turf.y + 1, iterating_turf.z)
+			if(!target_turf)
+				continue
+			var/area/target_area = target_turf.loc
+			if(target_area == src)
+				continue
+			if(!target_area.outdoors || target_area.underground)
+				continue
+			bitfield ^= bit_step
+
+		if(!bitfield)
+			continue
+		adjacent_day_night_turf_cache[iterating_turf] = list(DAY_NIGHT_TURF_INDEX_BITFIELD, DAY_NIGHT_TURF_INDEX_APPEARANCE)
+		adjacent_day_night_turf_cache[iterating_turf][DAY_NIGHT_TURF_INDEX_BITFIELD] = bitfield
+		RegisterSignal(iterating_turf, COMSIG_PARENT_QDELETING, .proc/clear_adjacent_turf)
+		if(iterating_turf.lighting_object)
+			iterating_turf.lighting_object.day_night_area = src
+
+	UNSETEMPTY(adjacent_day_night_turf_cache)
+
+/**
+ * Completely clears any adjacent turfs from the area while removing the effect.
+ */
+/area/proc/clear_adjacent_turfs()
+	for(var/turf/iterating_turf as anything in adjacent_day_night_turf_cache)
+		clear_adjacent_turf(iterating_turf)
+	adjacent_day_night_turf_cache = null
+
+/**
+ * Clears a signle turf from the adjacency turf cache.
+ * Arguments:
+ * * turf_to_clear - The turf we will unregister with and clear of any effects.
+ */
+/area/proc/clear_adjacent_turf(turf/turf_to_clear)
+	SIGNAL_HANDLER
+
+	turf_to_clear.underlays -= adjacent_day_night_turf_cache[turf_to_clear][DAY_NIGHT_TURF_INDEX_APPEARANCE]
+	adjacent_day_night_turf_cache -= turf_to_clear
+	UnregisterSignal(turf_to_clear, COMSIG_PARENT_QDELETING)
+
+/**
+ * A handler to set all adjacent turfs to the correct lighting.
+ */
+/area/proc/apply_day_night_turfs(datum/day_night_controller/incoming_controller)
+	SIGNAL_HANDLER
+
+	if(!incoming_controller)
+		return
+
+	for(var/turf/iterating_turf as anything in adjacent_day_night_turf_cache)
+		iterating_turf.underlays -= adjacent_day_night_turf_cache[iterating_turf][DAY_NIGHT_TURF_INDEX_APPEARANCE]
+		var/mutable_appearance/appearance_to_add = mutable_appearance(
+			'icons/effects/daynight_blend.dmi',
+			"[adjacent_day_night_turf_cache[iterating_turf][DAY_NIGHT_TURF_INDEX_BITFIELD]]",
+			DAY_NIGHT_LIGHTING_LAYER,
+			LIGHTING_PLANE,
+			incoming_controller.current_light_alpha,
+			RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+			)
+		appearance_to_add.color = incoming_controller.current_light_color
+		if(incoming_controller.current_luminosity)
+			iterating_turf.luminosity = incoming_controller.current_luminosity
+		iterating_turf.underlays += appearance_to_add
+		adjacent_day_night_turf_cache[iterating_turf][DAY_NIGHT_TURF_INDEX_APPEARANCE] = appearance_to_add
```

Comment:
...... how long is this list

Reply:
About ~1000 items long in total, generally speaking, it won't be updated much at all.

Reply:
jesus christ make this a subsystem fire

Reply:
I added a check tick as this is already called by the subsystem

Reply:
nevermind, how would you optimise this more?